### DISCUSSION TITLE
Integrating Next-Auth with existing NestJS backend

### DISCUSSION THREAD

**Discussion by chrber04** [nextauthjs/next-auth#8842] [Help]
### Summary

I currently have a custom NestJS backend that handles authentication using JWTs, with refresh tokens stored in a database and Passport.js for handling authentication. Given this setup, would there be any benefit in integrating Next-Auth into my application?

As far as I understand it, I would still need to create an API endpoint for Next-Auth, and so that would then need to call my NestJS backend. Would this add significant latency compared to making direct calls from the frontend to the NestJS backend? And even if so, would there still be benefits associated with still going with Next-Auth? 

Any insights would be greatly appreciated!

**Comment by kellenwiltshire** [comment]
Did you ever figure this out?
